randutil/crypto_test.go

```diff
@@ -45,3 +48,79 @@ func (s *cryptoRandutilSuite) TestCryptoToken(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(b, HasLen, 5)
 }
+
+var (
+	KernelTestUUID        = "1031319a-b661-4c01-aafa-6def8a118944"
+	KernelUUIDPathDefault = randutil.KernelUUIDPath
+)
+
+func (s *cryptoRandutilSuite) TestRandomKernelUUIDNoFile(c *C) {
+	tmpDir := c.MkDir()
+	uuidPath := filepath.Join(tmpDir, "no-file")
+
+	randutil.KernelUUIDPath = uuidPath
+	defer func() {
+		randutil.KernelUUIDPath = KernelUUIDPathDefault
+	}()
+
+	value, err := randutil.RandomKernelUUID()
+	c.Check(value, Equals, "")
+	c.Check(err, ErrorMatches, "*.no such file or directory")
+}
+
+func (s *cryptoRandutilSuite) TestRandomKernelUUIDNoPerm(c *C) {
+	tmpDir := c.MkDir()
+	uuidPath := filepath.Join(tmpDir, "no-perm")
+	err := ioutil.WriteFile(uuidPath, []byte(KernelTestUUID), 0)
```

Comment:
This will not work when the test runs as root, best to use something like "if os.Getuid() == 0 { c.Skip() }"